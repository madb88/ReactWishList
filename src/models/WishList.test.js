import { getSnapshot, onSnapshot, onPatch } from 'mobx-state-tree';
import { WishList, WishListItem } from "./WishList";
import { reaction } from 'mobx';

it('can create a instance of model', ()=> {
    const item = WishListItem.create({
        "name": "Test Name",
        "price": 28.23,
    })
    expect(item.price).toBe(28.23);
    expect(item.image).toBe('');
    item.changeName("Narnia");
    expect(item.name).toBe("Narnia");
})

it('can create a array of model', ()=> {
    const list = WishList.create({
        items: [
            {
                "name": "Test Name",
                "price": 28.23,
            }
        ]
    });
    expect(list.items.length).toBe(1);
    expect(list.items[0].price).toBe(28.23);
})

it('can add item to wishlist', ()=> {
    const list = WishList.create({
        items: []
    });
    const states = []
    onSnapshot(list, snapshot => {
        states.push(snapshot)
    })
    expect(list.items.length).toBe(0);

    list.add({
        "name": "Test Name",
        "price": 28.23,
        "image": ""
    });
    expect(list.items.length).toBe(1);

    expect(getSnapshot(list)).toMatchSnapshot()
    expect(getSnapshot(list)).toEqual({
        items:[{
            name: "Test Name",
            price: 28.23,
            image: ""
        }]
    })
    expect(states).toMatchSnapshot()
})


