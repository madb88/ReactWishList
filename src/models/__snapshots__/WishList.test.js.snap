// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`can add item to wishlist 1`] = `
Object {
  "items": Array [
    Object {
      "image": "",
      "name": "Test Name",
      "price": 28.23,
    },
  ],
}
`;

exports[`can add item to wishlist 2`] = `
Array [
  Object {
    "items": Array [
      Object {
        "image": "",
        "name": "Test Name",
        "price": 28.23,
      },
    ],
  },
]
`;
